db/queries/postsQueries.js

```diff
@@ -0,0 +1,70 @@
+const { db } = require("./index.js");
+
+const getAllPosts = (req, res, next) => {
+  db.any("SELECT * FROM posts")
+    .then(data => {
+      res.status(200).json({
+        status: "success",
+        message: "Received all POSTS",
+        data: data,
+      });
+    })
+    .catch(err => next(err));
+};
+
+const getSinglePost = (req, res, next) => {
+  let postId = parseInt(req.params.id);
+  db.one('SELECT * FROM posts WHERE id=$1', postId)
+    .then(data => {
+      res.status(200)
+        .json({
+          status: 'success',
+          message: "Received ONE POST!",
+          data: data,
+        })
+    })
+    .catch(err => next(err));
+}
+
+const createPost = (req, res, next) => {
+  db.none('INSERT INTO users(id,poster_id, body) VALUES(${id},${poster_id}, ${body})', req.body)
+    .then(data => {
+      res.status(200)
+        .json({
+          status: 'success',
+          message: 'Confirmed you have posted something '
+        })
+    })
+    .catch(err => next(err))
+}
+
```

Comment:
- Since your users.id is a serial primary key, you needn’t define what its value as it will automatically be assigned
- Your poster_id should be an integer, so you’ll need to use parseInt() or Number() on the req.body.poster_id